client/client-http/src/main/java/software/amazon/smithy/java/client/http/HttpErrorDeserializer.java

```diff
@@ -213,6 +214,41 @@ public CallException createError(
         }
     }
 
+    /**
+     * Create the error from response for restXml Protocol specifically.
+     *
+     * @param context      Context for the request.
+     * @param operation    Operation to create request for.
+     * @param typeRegistry TypeRegistry that can be used to create shapes.
+     * @param response     Response to deserialize.
+     * @return the deserialized error response.
+     */
+    public CallException createErrorForXml(
```

Comment:
HttpErrorDeserializer is not protocol specific, so it shouldn't know anything specific to XML here